### PR TITLE
Add fixes for accuracy test

### PR DESCRIPTION
**Pull request by avoraTT**
### Problem description
The accuracy test is currently broken on main, and not in CI.

### What's changed
- Fix accuracy test
- Add it to CI (TG Nightly)

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [x] [TG Nightly](https://github.com/tenstorrent/tt-metal/actions/runs/14219442841) Failed, but updated threshold

### CODE REVIEW

**Review by kpaigwar — APPROVED** [review]

**Comment by tt-rkim** [comment]
What's going on here

File: .github/workflows/tg-nightly-tests-impl.yaml [hunk 1]
```diff
@@ -20,6 +20,13 @@ jobs:
       matrix:
         test-group: [
           { name: "TG CCL tests", arch: wormhole_b0, cmd: "pytest tests/nightly/tg/ccl", timeout: 70, owner_id: ULMEPM2MA}, # Sean Nijjar
+          {
+            name: "Llama TG Accuracy Test",
+            arch: wormhole_b0,
+            cmd: "LLAMA_DIR=/mnt/MLPerf/tt_dnn-models/llama/Llama3.1-70B-Instruct/ FAKE_DEVICE=TG pytest models/demos/llama3_subdevices/tests/test_llama_accuracy.py",
+            timeout: 26,
```
Comment (ttmchiou) [inline .github/workflows/tg-nightly-tests-impl.yaml:27]: Why oddly 26 minutes?

File: models/demos/llama3_subdevices/tests/test_llama_accuracy.py [hunk 2]
```diff
@@ -53,10 +53,14 @@ def get_accuracy_thresholds(model_name: str, device_name: str, optimizations: Ll
 
 
 @torch.no_grad()
-@pytest.mark.timeout(900)
+@pytest.mark.timeout(1500)
```
Comment (ttmchiou) [inline models/demos/llama3_subdevices/tests/test_llama_accuracy.py:56]: we have a timeout in the workflow itself, is this timeout 25min timeout at the pytest level required?

File: .github/workflows/tg-nightly-tests-impl.yaml (hunk 1 above)
Comment (avoraTT) [reply .github/workflows/tg-nightly-tests-impl.yaml:27]: Test is suposed to take 20 mins, but sometimes the closing device takes longer.

File: models/demos/llama3_subdevices/tests/test_llama_accuracy.py (hunk 2 above)
Comment (avoraTT) [reply models/demos/llama3_subdevices/tests/test_llama_accuracy.py]: removed!

File: .github/workflows/tg-nightly-tests-impl.yaml (hunk 1 above)
Comment (avoraTT) [reply .github/workflows/tg-nightly-tests-impl.yaml]: updated it to be 25 min

**Review by ttmchiou — APPROVED** [review]

**Review by akirby-TT — APPROVED** [review]
